core/services/pipeline/common.go

```diff
@@ -60,14 +61,21 @@ var (
 
 type BaseTask struct {
 	outputTask Task
-	dotID      string `mapstructure:"-"`
-	Index      int32  `mapstructure:"index" json:"-" `
+	dotID      string        `mapstructure:"-"`
+	Index      int32         `mapstructure:"index" json:"-" `
+	Timeout    time.Duration `mapstructure:"timeout"`
```

Comment:
We went back and forth on this when writing the story. While it would be nice to have granularity on the task level, the story was asking for a job level configuration setting for all of its tasks. The problem here is that setting the configuration on every task is tedious and if you miss/forget one task then it will fall back to the node default. It seems more likely that users will want the same setting to timeout many concurrent fetches.

If you would like to leave this, it would be a nice feature. I think the order of precedence should be `task > job > node` configuration. Up to you, it's a little extra complexity, but it seems like we're most of the way there already. Either way, we should have a configuration setting on the job level.

Reply:
It was relatively straightforward to keep the per task specification, added a job level spec